lib/experimental/fonts-library/class-wp-fonts-library.php

```diff
@@ -0,0 +1,618 @@
+     * @return string|bool The relative path to the downloaded font asset. False if the download failed.
+     */
+    function download_asset( $src, $filename ) {
+        $file_path = path_join( $this->wp_fonts_dir, $filename );
+
+        // Checks if the file to be downloaded has a font mime type
+        if ( ! $this->has_font_mime_type( $file_path ) ) {
+            return false;
+        }
+
+        // Downloads the font asset or returns false
+        $temp_file = download_url( $src );
+        if ( is_wp_error( $temp_file ) ) {
+            @unlink( $temp_file );
+            return false;
+        }
+
+        // Moves the file to the fonts directory or return false
+        $renamed_file = rename( $temp_file, $file_path );
+        if ( ! $renamed_file ) {
+            @unlink( $temp_file );
+            return false;
+        }
+
+        // Cleans the temp file
+        @unlink( $temp_file );
+
+        // Returns the relative path to the downloaded font asset to be used as font face src
+        return "{$this->relative_fonts_path}{$filename}";
+    }
+
+    /**
+     * Generates a filename for a font face asset.
+     *
+     * Creates a filename for a font face asset using font family, style, weight and extension information.
+     *
+     * @param array $font_face The font face array containing 'fontFamily', 'fontStyle', and 'fontWeight' attributes.
+     * @param string $url The URL of the font face asset, used to derive the file extension.
+     * @param int $i Optional counter for appending to the filename, default is 0.
+     * @return string The generated filename for the font face asset.
+     */
+    function get_filename_from_font_face ( $font_face, $url, $i=0 ) {
+        $extension = pathinfo( $url, PATHINFO_EXTENSION );
+        $family = sanitize_title( $font_face['fontFamily'] );
+        $style = sanitize_title( $font_face['fontStyle'] );
+        $weight = sanitize_title( $font_face['fontWeight'] );
+        $filename = "{$family}_{$style}_{$weight}";
+        if ($i > 0) {
+            $filename = "{$filename}_{$i}";
+        }
+        return "{$filename}.{$extension}";
+    } 
+
+    /**
+     * Downloads font face assets.
+     *
+     * Downloads the font face asset(s) associated with a font face. It works with both single 
+     * source URLs and arrays of multiple source URLs.
+     *
+     * @param array $font_face The font face array containing the 'src' attribute with the source URL(s) of the assets.
+     * @return array The modified font face array with the new source URL(s) to the downloaded assets.
+     */
+    function download_font_face_assets ( $font_face ) {
+        $new_font_face = $font_face;
+        if ( is_array( $font_face['src'] ) ) {
+            $new_font_face['src'] = array();
+            $i;
+            foreach ( $font_face['src'] as $src ) {
+                !$i ? $i = 0 : $i++;
+                $filename = $this->get_filename_from_font_face( $font_face, $src, $i );
```

Comment:
I don't understand the need to check the value of `$i` with the ternary operator.
IMO this could be replaced with: 
```suggestion
            $i = 0;
            foreach ( $font_face['src'] as $src ) {              
                $filename = $this->get_filename_from_font_face( $font_face, $src, ++$i );
```